packages/grid_client/src/clients/kyc/client.ts

```diff
@@ -0,0 +1,180 @@
+   *
+   * @returns {Promise<void>}
+   * @private
+   */
+  private async setupKeyring() {
+    const keyring = new Keyring({ type: this.keypairType });
+    await waitReady();
+    this.keypair = keyring.addFromUri(this.mnemonic);
+    this.address = this.keypair.address;
+  }
+
+  /**
+   * Prepares the headers required for TFGrid KYC requests.
+   *
+   * This method generates a set of headers that include a timestamp-based challenge,
+   * a signed challenge using the user's key, and other necessary information.
+   *
+   * @returns {Promise<Record<string, string>>} A promise that resolves to an object containing the prepared headers.
+   *
+   */
+  private async prepareHeaders(): Promise<Record<string, string>> {
+    if (!this.keypair) await this.setupKeyring();
+    const timestamp = Date.now();
+    const challenge = stringToHex(`${this.apiDomain}:${timestamp}`);
+    const signedChallenge = this.keypair.sign(bytesFromHex(challenge));
+    const signedMsgHex = Buffer.from(signedChallenge).toString("hex");
+
+    const headers: KycHeaders = {
+      "content-type": "application/json",
+      "X-Client-ID": this.address,
+      "X-Challenge": challenge,
+      "X-Signature": signedMsgHex,
+    };
+    return headers as unknown as Record<string, string>;
+  }
+
+  /**
+   * Throws specific KYC-related errors based on the provided error message.
+   *
+   * @param errorMessage - The error message to evaluate.
+   * @throws {KycErrors.KycInvalidAddressError} If the error message indicates a malformed address.
+   * @throws {KycErrors.KycInvalidChallengeError} If the error message indicates a malformed or bad challenge.
+   * @throws {KycErrors.KycInvalidSignatureError} If the error message indicates a malformed or bad signature.
+   * @returns {undefined} If the error message does not match any known error patterns.
+   */
+  private throwKycError(error: Error, messagePrefix: string) {
+    if (!(error instanceof AxiosError)) return new KycBaseError(error.message);
+    const { response, status } = error as AxiosError;
+    if (response?.data) error.message = (response?.data as { error: string })?.error || error.message;
+    const errorMessage = formatErrorMessage(messagePrefix, error);
+    switch (true) {
+      case status === HttpStatusCode.BadRequest:
+        return new KycErrors.BadRequest(`${errorMessage}. Please contact support.`);
+
+      case status === HttpStatusCode.Unauthorized:
+        return new KycErrors.Unauthorized(`${errorMessage}. Please contact support.`);
+
+      case status === HttpStatusCode.NotFound:
+        return new KycErrors.Unverified(errorMessage);
+
+      case status === HttpStatusCode.TooManyRequests:
+        return new KycErrors.RateLimit(errorMessage);
+
+      case status === HttpStatusCode.Conflict:
+        return new KycErrors.AlreadyVerified(errorMessage);
+
+      case status === HttpStatusCode.PaymentRequired:
+        return new InsufficientBalanceError(errorMessage);
+    }
```

Comment:
And else?

Reply:
default case was removed by mistake, added now